src/s3ql/metadata.py

```diff
@@ -133,7 +133,7 @@ def cycle_metadata(backend, keep=10):
         pass
     cycle_fn("s3ql_metadata_new", "s3ql_metadata")
 
-    if cycle_fn is backend.copy:
+    if not backend.has_native_rename:
         backend.delete('s3ql_metadata_new')
```

Comment:
If `cycle_fn` is not `backend.copy`, then it is `backend.rename`, so there should be no `s3ql_metadata_new` object (it has been renamed). Thus the current code seems ok.

Looking at your proposed change, note that if `not backend.has_native_rename`, then `cycle_fn is backend.copy` (per line 115-118), so your change doesn't actually change anything in practice.